File: android/src/oldarch/java/com/callkeeper/CallKeeperSpec.java
package com.callkeeper;

import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.bridge.ReactContextBaseJavaModule;
import com.facebook.react.bridge.Promise;
import com.facebook.react.bridge.ReadableMap;

abstract class CallKeeperSpec extends ReactContextBaseJavaModule {
  CallKeeperSpec(ReactApplicationContext context) {
    super(context);
  }

  public abstract void setup(ReadableMap options, Promise promise);
  public abstract void displayIncomingCall(String callUUID, String handle, String localizedCallerName, String handleType, boolean hasVideo, Promise promise);
  public abstract void startCall(String callUUID, String handle, String contactIdentifier, String handleType, boolean hasVideo, Promise promise);
  public abstract void endCall(String callUUID, Promise promise);
  public abstract void endAllCalls(Promise promise);
  public abstract void answerIncomingCall(String callUUID, Promise promise);
  public abstract void rejectCall(String callUUID, Promise promise);
  public abstract void setMutedCall(String callUUID, boolean muted, Promise promise);
  public abstract void setOnHold(String callUUID, boolean onHold, Promise promise);
  public abstract void reportConnectedOutgoingCall(String callUUID, Promise promise);
  public abstract void reportEndCallWithUUID(String callUUID, int reason, Promise promise);
  public abstract void updateDisplay(String callUUID, String displayName, String handle, Promise promise);
  public abstract void checkPermissions(Promise promise);
  public abstract void checkIsInManagedCall(Promise promise);
  public abstract void setAvailable(boolean available, Promise promise);
  public abstract void setCurrentCallActive(String callUUID, Promise promise);
  public abstract void backToForeground(Promise promise);
}

File: android/src/oldarch/kotlin/com/callkeeper/CallKeeperSpec.kt
package com.callkeeper

import com.facebook.react.bridge.Promise
import com.facebook.react.bridge.ReactApplicationContext
import com.facebook.react.bridge.ReactContextBaseJavaModule
import com.facebook.react.bridge.ReadableMap

abstract class CallKeeperSpec(context: ReactApplicationContext) : ReactContextBaseJavaModule(context) {
    abstract fun setup(options: ReadableMap, promise: Promise)
    abstract fun displayIncomingCall(
        callUUID: String,
        handle: String,
        localizedCallerName: String?,
        handleType: String?,
        hasVideo: Boolean?,
        promise: Promise
    )
    abstract fun startCall(
        callUUID: String,
        handle: String,
        contactIdentifier: String?,
        handleType: String?,
        hasVideo: Boolean?,
        promise: Promise
    )
    abstract fun endCall(callUUID: String, promise: Promise)
    abstract fun endAllCalls(promise: Promise)
    abstract fun answerIncomingCall(callUUID: String, promise: Promise)
    abstract fun rejectCall(callUUID: String, promise: Promise)
    abstract fun setMutedCall(callUUID: String, muted: Boolean, promise: Promise)
    abstract fun setOnHold(callUUID: String, onHold: Boolean, promise: Promise)
    abstract fun reportConnectedOutgoingCall(callUUID: String, promise: Promise)
    abstract fun reportEndCallWithUUID(callUUID: String, reason: Double, promise: Promise)
    abstract fun updateDisplay(callUUID: String, displayName: String, handle: String, promise: Promise)
    abstract fun checkPermissions(promise: Promise)
    abstract fun checkIsInManagedCall(promise: Promise)
    abstract fun setAvailable(available: Boolean, promise: Promise)
    abstract fun setCurrentCallActive(callUUID: String, promise: Promise)
    abstract fun backToForeground(promise: Promise)
}

